core/node/consensus/src/vm.rs

```diff
@@ -20,34 +18,26 @@ use crate::{abi, storage::ConnectionPool};
 #[derive(Debug)]
 pub(crate) struct VM {
     pool: ConnectionPool,
-    setup_args: TxSetupArgs,
-    limiter: VmConcurrencyLimiter,
+    options: OneshotEnvParameters<CallOrExecute>,
+    executor: MainOneshotExecutor,
 }
 
 impl VM {
     /// Constructs a new `VM` instance.
     pub async fn new(pool: ConnectionPool) -> Self {
         Self {
             pool,
-            setup_args: TxSetupArgs {
-                execution_mode: TxExecutionMode::EthCall,
-                operator_account: AccountTreeId::default(),
-                fee_input: BatchFeeInput::sensible_l1_pegged_default(),
-                base_system_contracts: scope::wait_blocking(
-                    MultiVMBaseSystemContracts::load_eth_call_blocking,
-                )
-                .await,
-                caches: PostgresStorageCaches::new(1, 1),
-                validation_computational_gas_limit: u32::MAX,
-                chain_id: L2ChainId::default(),
-                whitelisted_tokens_for_aa: vec![],
-                enforced_base_fee: None,
-            },
-            limiter: VmConcurrencyLimiter::new(1).0,
+            options: OneshotEnvParameters::for_execution(
+                L2ChainId::default(),
+                AccountTreeId::default(),
```

Comment:
I *think* L2 chain ID and the fee account address don't matter for calls, but probably makes sense to double-check. I know for certain that L2 chain ID passed to `multivm` does matter for transactions because it's used for signature verification.

Reply:
Added a comment above, so it would be easier to investigate if this assumption turns out to be wrong.